pkg/proxy/proxy_test.go

```diff
@@ -123,524 +123,611 @@ func (s *TestProxySuite) TestProxy() {
-					resp, err := http.DefaultClient.Do(req)
-
-					// then
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), resp)
-					defer resp.Body.Close()
-					assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
-					s.assertResponseBody(resp, "invalid bearer token: unable to extract userID from token: token contains an invalid number of segments")
-				})
-
-				s.Run("unauthorized if can't extract userID from a valid token", func() {
-					// when
-					req, err := http.NewRequest("GET", "http://localhost:8081/api/mycoolworkspace/pods", nil)
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), req)
-					userID, err := uuid.NewV4()
-					require.NoError(s.T(), err)
-					req.Header.Set("Authorization", "Bearer "+s.token(userID, authsupport.WithSubClaim("")))
-					resp, err := http.DefaultClient.Do(req)
-
-					// then
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), resp)
-					defer resp.Body.Close()
-					assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
-					s.assertResponseBody(resp, "invalid bearer token: unable to extract userID from token: token does not comply to expected claims: subject missing")
-				})
-
-				s.Run("unauthorized if workspace context is invalid", func() {
-					// when
-					req := s.request()
-					req.URL.Path = "http://localhost:8081/workspaces/myworkspace" // invalid workspace context
-					require.NotNil(s.T(), req)
-
-					// when
-					resp, err := http.DefaultClient.Do(req)
-
-					// then
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), resp)
-					defer resp.Body.Close()
-					assert.Equal(s.T(), http.StatusBadRequest, resp.StatusCode)
-					s.assertResponseBody(resp, "unable to get workspace context: workspace request path has too few segments '/workspaces/myworkspace'; expected path format: /workspaces/<workspace_name>/api/...")
-				})
-
-				s.Run("internal error if get accesses returns an error", func() {
-					// given
-					req := s.request()
-					fakeApp.Accesses = map[string]*access.ClusterAccess{}
-					fakeApp.Err = errors.New("some-error")
-
-					// when
-					resp, err := http.DefaultClient.Do(req)
-
-					// then
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), resp)
-					defer resp.Body.Close()
-					assert.Equal(s.T(), http.StatusInternalServerError, resp.StatusCode)
-					s.assertResponseBody(resp, "unable to get target cluster: some-error")
-				})
-
-				s.Run("internal error if accessing incorrect url", func() {
-					// given
-					req := s.request()
-					req.URL.Path = "http://localhost:8081/metrics"
-					require.NotNil(s.T(), req)
-
-					// when
-					resp, err := http.DefaultClient.Do(req)
-
-					// then
-					require.NoError(s.T(), err)
-					require.NotNil(s.T(), resp)
-					defer resp.Body.Close()
-					assert.Equal(s.T(), http.StatusInternalServerError, resp.StatusCode)
-				})
-			})
+			s.checkPlainHTTPErrors(fakeApp)
+			s.checkWebsocketsError()
+			s.checkWebLogin()
```

Comment:
Please review the `s.checkWebLogin()` function with the new tests. The rest of the change here is just to make the linter happy. I moved other tests to separate functions as is without changing anything there.